addons/mail/models/ir_attachment.py

```diff
@@ -10,6 +10,15 @@
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    @api.model
+    def check(self, mode, values=None):
+        super().check(mode, values=values)
+        if mode != 'unlink' or not self or self.env.is_admin():
+            return
+        authors = self.env['mail.message'].sudo().search([('attachment_ids', 'in', self.ids)]).author_id
```

Comment:
can we add a test for the tricky scenario where the same attachment is on the 2 messages ?
(the current code support it already I think)